src/fheroes2/maps/map_object_info.cpp

```diff
@@ -4487,6 +4487,49 @@ namespace
             objects.emplace_back( std::move( object ) );
         }
 
+        // Observation Tower, generic terrain.
+        {
+            Maps::ObjectInfo object{ MP2::OBJ_OBSERVATION_TOWER };
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 235U, fheroes2::Point{ 0, 0 }, MP2::OBJ_OBSERVATION_TOWER, Maps::OBJECT_LAYER );
+
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 199U, fheroes2::Point{ -2, 0 }, MP2::OBJ_NONE, Maps::SHADOW_LAYER );
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 200U, fheroes2::Point{ -1, 0 }, MP2::OBJ_NONE, Maps::SHADOW_LAYER );
+
+            object.topLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 198U, fheroes2::Point{ 0, -1 }, MP2::OBJ_NON_ACTION_OBSERVATION_TOWER );
+
+            objects.emplace_back( std::move( object ) );
+        }
+
+        // Observation Tower, desert terrain.
+        {
+            Maps::ObjectInfo object{ MP2::OBJ_OBSERVATION_TOWER };
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 236U, fheroes2::Point{ 0, 0 }, MP2::OBJ_OBSERVATION_TOWER, Maps::OBJECT_LAYER );
+
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 199U, fheroes2::Point{ -2, 0 }, MP2::OBJ_NONE, Maps::SHADOW_LAYER );
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 200U, fheroes2::Point{ -1, 0 }, MP2::OBJ_NONE, Maps::SHADOW_LAYER );
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 237U, fheroes2::Point{ 1, 0 }, MP2::OBJ_NON_ACTION_OBSERVATION_TOWER,
+                                                  Maps::BACKGROUND_LAYER );
+
+            object.topLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 198U, fheroes2::Point{ 0, -1 }, MP2::OBJ_NON_ACTION_OBSERVATION_TOWER );
+
+            objects.emplace_back( std::move( object ) );
+        }
+
+        // Observation Tower, snow terrain.
+        {
+            Maps::ObjectInfo object{ MP2::OBJ_OBSERVATION_TOWER };
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 238U, fheroes2::Point{ 0, 0 }, MP2::OBJ_OBSERVATION_TOWER, Maps::OBJECT_LAYER );
+
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 199U, fheroes2::Point{ -2, 0 }, MP2::OBJ_NONE, Maps::SHADOW_LAYER );
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 200U, fheroes2::Point{ -1, 0 }, MP2::OBJ_NONE, Maps::SHADOW_LAYER );
+            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 239U, fheroes2::Point{ 1, 0 }, MP2::OBJ_NON_ACTION_OBSERVATION_TOWER,
+                                                  Maps::BACKGROUND_LAYER );
```

Comment:
Yes, there is a small pile of snow, but it does not block vertical movement on the right side of the tower.
![изображение](https://github.com/user-attachments/assets/7399f7ab-1e68-488a-9b0d-efe9a42f0c39)
Should we move the right part of this object to the `TERRAIN_LAYER`?
```suggestion
            object.groundLevelParts.emplace_back( MP2::OBJ_ICN_TYPE_OBJNMUL2, 239U, fheroes2::Point{ 1, 0 }, MP2::OBJ_NONE, Maps::TERRAIN_LAYER );
```

Reply:
Hi @Districh-ru , it is intentional and also an action object cannot have terrain layer parts. An assertion at line 4955 (`assert( info.objectType == MP2::OBJ_NONE );`) will blow up because of this.

Reply:
Hi, @ihhub this assertion blows up if the `TERRAIN_LAYER` object part is not `MP2::OBJ_NONE`. So this change will not break the assertion:
![изображение](https://github.com/user-attachments/assets/1354aa2e-4f3c-4f7a-bae9-971c4abde6db)

IMHO the current passablilty is strange - the pile does not block the vertical movement, it can block the horizontal movement from the center to the right tile. But it is mine POV. :)

@Branikolog, @oleg-derevenetz and @zenseii, what do you think about these towers passability?
Should it be like for all other towers? Or the vertical movement at the right part of the tower should be blocked if there is a snow/sand pile part in the lower tile?

Reply:
I really don't think that the same object should have different passability on different terrains.

Reply:
Hi @Districh-ru and @oleg-derevenetz , I've changed the logic to match the original tower passability.